### PR TITLE
fix: correct mapping for properties object for identify call

### PR DESCRIPTION
**Pull request by ujjwal-ab**
## Description of the change

- fix incorrect properties mapping 
- add support of `make email or phone as the primary identifier`,  to track call

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: src/v0/destinations/klaviyo/transform.js [hunk 1]
```diff
@@ -186,8 +189,9 @@ const trackRequestHandler = (message, category, destination) => {
       ...attributes.properties,
       ...populateCustomFieldsFromTraits(message),
     };
-    attributes.profile = createCustomerProperties(message);
   }
+  // Map user properties to profile object
+  attributes.profile = createCustomerProperties(message, destination.Config);
```
Comment (krishna2020) [inline src/v0/destinations/klaviyo/transform.js:194]: Is it customerProperties or customProperties ?

Comment (ujjwal-ab) [reply src/v0/destinations/klaviyo/transform.js:194]: These are customer's profile properties. Eg. Email, firstname, lastname etc

File: src/v0/destinations/klaviyo/util.js [hunk 2]
```diff
@@ -135,12 +135,25 @@ const subscribeUserToList = (message, traitsInfo, destination) => {
...
+  const { enforceEmailAsPrimary } = Config;
   let customerProperties = constructPayload(
     message,
     MAPPING_CONFIG[CONFIG_CATEGORIES.PROFILE.name],
   );
-  customerProperties.$id = getFieldValueFromMessage(message, 'userId');
+  if (!enforceEmailAsPrimary) {
+    customerProperties.$id = getFieldValueFromMessage(message, 'userId');
+  }
+  if (enforceEmailAsPrimary) {
```
[the remark this answers is not in the document]
Comment (ujjwal-ab) [reply src/v0/destinations/klaviyo/util.js]: Updated

File: src/v0/destinations/klaviyo/util.js [hunk 3]
```diff
@@ -135,12 +135,25 @@ const subscribeUserToList = (message, traitsInfo, destination) => {
...
+    if (!customerProperties.$email && !customerProperties.$phone_number) {
+      throw new InstrumentationError('None of email and phone are present in the payload');
+    } else {
```
[the remark this answers is not in the document]
Comment (ujjwal-ab) [reply src/v0/destinations/klaviyo/util.js]: removed

**Review by krishna2020 — APPROVED** [review]

**Review by utsabc — APPROVED** [review]

**Pull request merged**